cmd/d8s/delete.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"net/http"
+
+	"github.com/google/go-containerregistry/authn"
+	"github.com/google/go-containerregistry/name"
+	"github.com/google/go-containerregistry/v1/remote"
+	"github.com/google/subcommands"
+)
+
+type deleteCmd struct{}
+
+func (*deleteCmd) Name() string             { return "delete" }
+func (*deleteCmd) Synopsis() string         { return "Deletes a reference from its registry" }
+func (*deleteCmd) Usage() string            { return "delete <image>" }
+func (*deleteCmd) SetFlags(f *flag.FlagSet) {}
+
+func (*deleteCmd) Execute(ctx context.Context, f *flag.FlagSet, _ ...interface{}) subcommands.ExitStatus {
+	if len(f.Args()) != 1 {
+		return subcommands.ExitUsageError
+	}
+	ref := f.Args()[0]
+
+	r, err := name.ParseReference(ref, name.WeakValidation)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	auth, err := authn.DefaultKeychain.Resolve(r.Context().Registry)
+	if err != nil {
```

Comment:
I could see us wanting a mechanism for overriding this default behavior, e.g. to support `imagePullSecrets` or force a particular cred helper.

Reply:
That makes sense. I can add a command-wide flag to override the auth method, so it's applicable to all commands.

Does `authn` support `imagePullSecrets` today? Could `DefaultKeychain` fallback to check `imagePullSecrets`? (These are questions for a future change, either way)

Reply:
Not yet.  I want to add a mode to these to support K8s-compatible auth resolution.